### ISSUE TITLE
cant login to anything

### ISSUE THREAD

**Issue by HexorCatZ** [BlackArch/blackarch#2835]
Type: Discussion

## Discussion
On 22 Aug 2020
i had a problem (im not sure if it was my faults or a bug).
Just after update (pacman -Syu) and reboot, i cant login back.

lxdm (is it lxdm or else? The default that came with it) will just react as i type the wrong password.
enter user => enter pass => ask for username again

On tty, theres a "login incorrect" messages after key in the password and before it refresh to ask for username again.

I tried to troubleshoot it by using a live arch and change the password but no different.
ssh also react as if i give it a wrong password.

Tried reinstall the kernel(pacman -Sy linux linux-firmware linux-headers) (maybe it'll help) with no luck.
have tried reinstall pam and pambase too.

Oh and this is the second time happened to me.
The first one is on a different machine with headless setup.
And i just reinstall everything as i thought i did something wrong.
just a morning after, this happened to my main laptop.

If someone knows the solution to this or someone facing the same problem, please let me know.
So that i dont have to blame myself. :D

**Comment by noraj** [comment]
The explanation is way too blurry, I don't think we can help here.

**Comment by HexorCatZ** [comment]
ops, sorry. my bad.
okay, let me try again.

the stories : 
2 days ago, at night, i did an update on my computer using `pacman -Syu`.
but after reboot, i cant login to it.
i did troubleshot it for an hour or two with no luck.
then i decides to reinstall everything.

and while waiting for the installation on my PC, i update my laptop. 😅 
morning after, my laptop is also facing the same problem.

the problem : 
it started as usual until at the login screen.
it act as i type the wrong username and password which i definitely use to login everyday.
from the tty shell, it keeps telling me that "login incorrect" even when i give a correct password.
same as from ssh

attempt troubleshoot :  
tried to change my password from live boot of arch.
tried `pacman -S linux linux-firmware linux-headers`
tried `pacman -S pam pambase`
tried `pacman -Syu` (and hoping that it will magically works)

update : 
tho i was able to boot up to rescue mode this morning before i go to work.
by booting to live arch, change my GRUB_TIMEOUT=0.1 to 3 and add single by pressing 'e' in grub menu
and my root password works there too but not after a normal boot. weird.

im not too sure where to find a log in arch as its not in /var/log/ or im just stupid at it. 😞

**Comment by ikstream** [comment]
please check if there is a `/etc/pam.d/system-login.pacnem` if so please change your `/etc/pam.d/system-login` accordingly

**Comment by HexorCatZ** [comment]
Okay, when you(ikstream) asked me to check on pam related files, i know it has to do with pam.

And i just realized that arch does not save logs in text but in journal instead.
And journalctl looks more like dmesg which im already used to.
Now i feel stupid that i dont know about it.

By reading a pam related logs (`journalctl -g pam`) i found out that theres a log saying...
(on phone and retyping the log. Sorry if i had type something wrong)

`Aug 22 22:02:26 catz lxdm-session[13078]: PAM unable to dlopen(/usr/lib/security/pam_tally.so): /usr/lib/security/pam_tally.so: cannot open shared object file: No such file or directory`
`Aug 22 22:02:26 catz lxdm-session[13078]: PAM adding faulty module: /usr/lib/security/pam_tally.so`

I think this has to do with it as the error first popped up on the date when i had this problems.

**Comment by HexorCatZ** [comment]
i tried removing pam_tally.so from `/etc/pam.d/system-login` and it works.
im typing this using that laptop. :smile: 
but i want some advice, is it a good idea to remove it?
if you say yes, ive consider done and close this issue

**Comment by HexorCatZ** [comment]
okay, as mention in [here](https://man7.org/linux/man-pages/man8/pam_tally.8.html
)
```
pam_tally has several limitations, which are solved with pam_tally2.
For this reason pam_tally is deprecated and will be removed in a
future release.
```
which means it already been removed and I have to upgrade it to pam_tally2 later on.
thank you guys for the help. i really apperciarte it.

ill close this issue

**Comment by samspei0l** [comment]
I tried everything here but still can't login to my blackarch.

**Comment by ikstream** [comment]
Then reopen your issue and provide more information. Don't cross post. use chroot and show us your logs
